From a68735ff53b45e7729dd112d1b8c56d2ece6016a Mon Sep 17 00:00:00 2001
From: Alex Vd <keromvp@gmail.com>
Date: Fri, 17 Jan 2025 21:55:35 +0300
Subject: [PATCH 1/1] fix write

Signed-off-by: Alex Vd <keromvp@gmail.com>
---
 include/crow/http_connection.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/crow/http_connection.h b/include/crow/http_connection.h
index 908b8ec8..34e0b637 100644
--- a/include/crow/http_connection.h
+++ b/include/crow/http_connection.h
@@ -103,7 +103,7 @@ namespace crow
                 buffers_.clear();
                 static std::string expect_100_continue = "HTTP/1.1 100 Continue\r\n\r\n";
                 buffers_.emplace_back(expect_100_continue.data(), expect_100_continue.size());
-                do_write();
+                do_write_sync(buffers_);
             }
         }
 
@@ -408,7 +408,7 @@ namespace crow
                 res_body_copy_.swap(res.body);
                 buffers_.emplace_back(res_body_copy_.data(), res_body_copy_.size());
 
-                do_write();
+                do_write_sync(buffers_);
 
                 if (need_to_start_read_after_complete_)
                 {
-- 
2.45.2

