diff --git a/include/crow/http_response.h b/include/crow/http_response.h
index e235025e..e644678d 100644
--- a/include/crow/http_response.h
+++ b/include/crow/http_response.h
@@ -20,6 +20,7 @@
 #include "crow/logging.h"
 #include "crow/mime_types.h"
 #include "crow/returnable.h"
+#include "json.h"
 
 
 namespace crow
@@ -318,6 +319,16 @@ namespace crow
             }
         }
 
+        void sendJSON(const nlohmann::json& data)
+        {
+            std::string response = data.dump();
+
+            add_header("Access-Control-Allow-Origin", "*");
+            add_header("Content-Type", "text/html");
+            write(response);
+            end();
+        }
+
     private:
         bool completed_{};
         std::function<void()> complete_request_handler_;
